ip-messaging/users/token-generation-server/token-generation-server.5.x.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+using Twilio;
+using Twilio.Auth;
+using Twilio.Jwt.AccessToken;
+using Faker;
+
+namespace TwilioIpMessaging.Controllers
+{
+    public class TokenController : Controller
+    {
+        // GET: /token
+        public ActionResult Index(string device)
+        {
+            // Load Twilio configuration from Web.config
+            var accountSid = ConfigurationManager.AppSettings["TwilioAccountSid"];
+            var apiKey = ConfigurationManager.AppSettings["TwilioApiKey"];
+            var apiSecret = ConfigurationManager.AppSettings["TwilioApiSecret"];
+            var ipmServiceSid = ConfigurationManager.AppSettings["TwilioIpmServiceSid"];
+
+            // Create a random identity for the client
+            var identity = Internet.UserName();
+
+            // Create an IP messaging grant for this token
+            var grant = new IpMessagingGrant();
+            grant.EndpointId = $"TwilioChatDemo:{identity}:{device}";
+            grant.ServiceSid = ipmServiceSid;
+
+            var grants = new HashSet<IGrant>
+            {
+              { grant }
```

Comment:
Although the extra curly braces are not an error, they introduced an additional scope. Would you mind removing them?